packages/node/src/handlers.ts

```diff
@@ -173,18 +174,11 @@ export function requestHandler(
     next: (error?: any) => void,
   ): void {
     if (options && options.flushTimeout && options.flushTimeout > 0) {
-      // eslint-disable-next-line @typescript-eslint/unbound-method
-      const _end = res.end;
-      res.end = function (chunk?: any | (() => void), encoding?: string | (() => void), cb?: () => void): void {
-        void flush(options.flushTimeout)
-          .then(() => {
-            _end.call(this, chunk, encoding, cb);
-          })
-          .then(null, e => {
-            __DEBUG_BUILD__ && logger.error(e);
-            _end.call(this, chunk, encoding, cb);
-          });
-      };
+      onFinished(res, () => {
```

Comment:
```js
function isFinished(msg) {
  var socket = msg.socket;

  if (typeof msg.finished === 'boolean') {
    // OutgoingMessage
    return Boolean(msg.finished || (socket && !socket.writable));
  }

  if (typeof msg.complete === 'boolean') {
    // IncomingMessage
    return Boolean(msg.upgrade || !socket || !socket.readable || (msg.complete && !msg.readable));
  }

  // don't know
  return undefined;
}
```

```js
  // finished on first message event
  eeMsg = eeSocket = first([[msg, 'end', 'finish']], onFinish);

  function onSocket(socket) {
    // remove listener
    msg.removeListener('socket', onSocket);

    if (finished) return;
    if (eeMsg !== eeSocket) return;

    // finished on first socket event
    eeSocket = first([[socket, 'error', 'close']], onFinish);
  }
```
I believe there is not much difference. If we are certain that `res.once('finish', () => {})` will work as expected, then there is no need to introduce `onFinished`. 
1. `onFinished` includes some compatibility code for older versions of Node.js. 
2. Additionally, when the isFinished property of res is true, it can also trigger the callback function registered with onFinished. 
3. `onFinished` register more event from http response object, like `error`, `close`, `end`, `finish`

above is what `onFinished` do

In my opinion, using `res.once('finish', () => {})` is a more concise and maintainable approach. 

On the other hand, using onFinished may carry more significance in certain cases. 

if have any idea,  we can discuss it further.  If necessary, we can remove onFinished and directly use the standard event in Node.js.

Reply:
> `onFinished` includes some compatibility code for older versions of Node.js.

That's the one thing that might be important. Do you know what the minimum version for `res.once('finish')` is? I tried looking it up in the Node docs but couldn't find much. (Seems like `once` isn't even documented, unless I'm missing something). What we can see is that the `'finish'` event was introduced in 0.3.6 so that should be no problem. **Whatever we do here needs to support Node 8 and higher.**

> In my opinion, using res.once('finish', () => {}) is a more concise and maintainable approach.

I agree. If it works for Node 8+, I think we should use it instead of adding `onFinished`. If it turns out that this doesn't work, we can still come back to `onFinished`.

Reply:
@jiangbo0216 would you mind giving `res.once('finish')` a try?

Reply:
ok, I agree with you

Reply:
@Lms24
Could you help review this when you have time? Thank you.